### PR TITLE
Upgrade popper to v2

### PR DESCRIPTION
**Pull request by veegandhi**
## Description

Upgrade `react-popper` to v2, it's mostly compatible with the current usage of the component `<Popover>`. 

Except for one change as below images:

Screen Shot 2021-02-22 at 6 24 13 pm

Also, refined `<HoverDropdownMenu />`and `<TextEllipsis />`

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/components/Popover/index.jsx [hunk 1]
```diff
@@ -2,132 +2,120 @@
...
+  //     { name: 'arrow', options: { element: arrowElement } },
+  //     { name: 'preventOverflow', options: { enabled: true, boundariesElement: getBoundedContainer() } },
+  //   ],
+  //   placement: props.placement,
+  //   wrapperStyles: props.wrapperStyles,
+  // });
+
+  const popoverElement = isPopoverOpen
+    ? ReactDOM.createPortal(
+        <Popper
```
Comment (veegandhi) [inline src/components/Popover/index.jsx:58]: Come to think we might just use `Popover.WithRef` internally here.

File: src/components/Popover/Popper.jsx [hunk 2]
```diff
@@ -45,70 +45,94 @@ export const renderArrowStyles = (placement, arrowStyles, container = null) => {
...
-            className="aui--popover-arrow"
-            data-placement={placement}
-            ref={arrowProps.ref}
-            style={{ ...arrowProps.style, ...calculatedArrowStyles }}
-          />
+      <div className="aui--popover-container">
+        {title ? <div className="popover-title">{title}</div> : null}
+        <div className="popover-content">
+          {/* {_.isFunction(popoverContent) ? popoverContent({ scheduleUpdate }) : popoverContent} */}
+          {_.isFunction(popoverContent) ? popoverContent({ scheduleUpdate: update }) : popoverContent}
```
Comment (veegandhi) [inline src/components/Popover/Popper.jsx:102]: **_Note to self_**: `scheduledUpdate` will need to be extensively tested with grid.

**Review by vinteo — APPROVED** [review]

**Review by pphminions — APPROVED** [review]
LGTM 🟠

File: src/components/Popover/WithRef.jsx [hunk 3]
```diff
@@ -28,47 +28,43 @@ const WithRef = ({
...
+      modifiers={modifiers}
+      popperRef={popperRef}
+    />,
+    boundariesElement
   );
 };
 
 WithRef.propTypes = {
-  refElement: PropTypes.instanceOf(Element).isRequired,
+  refElement: PropTypes.instanceOf(Element),
```
Comment (superpower1) [inline src/components/Popover/WithRef.jsx:50]: Why is refElement no longer required?

Comment (Harris-YWang) [reply src/components/Popover/WithRef.jsx:50]: After updating, the `refElement` is initally `null` due to `React.createRef(null)` or `React.useRef()`, and it's managed by react itself

**Comment by bchew** [comment]
Merging :rocket:

**Pull request merged**